psw/urts/linux/isgx_user.h

```diff
@@ -109,11 +109,11 @@ enum sgx_page_flags {
 #define SGX_IOC_ENCLAVE_SET_ATTRIBUTE \
 	_IOW(SGX_MAGIC, 0x03, struct sgx_enclave_set_attribute)
 #define SGX_IOC_PAGE_MODP \
-	_IOWR(SGX_MAGIC, 0x04, struct sgx_page_modp)
+	_IOWR(SGX_MAGIC, 0x05, struct sgx_page_modp)
 #define SGX_IOC_PAGE_MODT \
-	_IOWR(SGX_MAGIC, 0x05, struct sgx_page_modt)
+	_IOWR(SGX_MAGIC, 0x06, struct sgx_page_modt)
 #define SGX_IOC_PAGE_REMOVE \
-    _IOWR(SGX_MAGIC, 0x06, struct sgx_page_remove)
+    _IOWR(SGX_MAGIC, 0x07, struct sgx_page_remove)
 
```

Comment:
This change is to align current patch under review in LKML:https://patchwork.kernel.org/project/intel-sgx/cover/cover.1638381245.git.reinette.chatre@intel.com/
If you still use the branch from kernel repo mentioned in README, do NOT use the change.

Note you also need enable RWX for EAUG pages to be able to change permissions arbitrarily. (this is not supported in LKML patches.) 
Add this change for kernel after applying all EDMM patches.

 diff --git a/arch/x86/kernel/cpu/sgx/encl.c b/arch/x86/kernel/cpu/sgx/encl.c
index 37203da382f8..8f50a5c4d125 100644
--- a/arch/x86/kernel/cpu/sgx/encl.c
+++ b/arch/x86/kernel/cpu/sgx/encl.c
@@ -165,7 +165,7 @@ static vm_fault_t sgx_encl_eaug_page(struct vm_area_struct *vma,
         * TBD: Interface with user space policy to support max permissions
         * of RWX.
         */
-       prot = PROT_READ | PROT_WRITE;
+       prot = PROT_READ | PROT_WRITE | PROT_EXEC;
        encl_page->vm_run_prot_bits = calc_vm_prot_bits(prot, 0);
        encl_page->vm_max_prot_bits = encl_page->vm_run_prot_bits;

Reply:
LKML published patches +above RWX temporary patch are hosted here now:  https://github.com/rchatre/linux/commits/sgx/sgx2_submitted_v1_plus_rwx

Updated README